### PR TITLE
Configure TravisCI builds

### LINKED ISSUE #4: Configure CI
Since the project is now public, I think it would be nice to have some CI set up to run tests on each PR. I see that Phoenix itself is using TravisCI, should we use it for this project as well?

/cc @josevalim @chrismccord

### PR DESCRIPTION
**Pull request by arkgil**
This PR adds the `.travis.yml` file.

Closes #4

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+elixir:
+  - 1.4
+  - 1.5
+  - 1.6
+otp_release:
+  - 19.3
+  - 20.3
+  - 21.0
+matrix:
+  exclude:
```
Comment (josevalim) [inline .travis.yml]: I tend to prefer explicit includes as it gets very messy as we add new Elixir and OTP versions. Thoughts?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+language: elixir
+elixir:
+  - 1.4
```
Comment (josevalim) [inline .travis.yml]: Can we run on earlier versions?

Comment (josevalim) [reply .travis.yml]: Nevermind, I see the rationale for v1.4. :)

Comment (arkgil) [reply .travis.yml]: I'll try with Elixir 1.3.

What about OTP releases? Should we test on 18.3 as well?

Comment (josevalim) [reply .travis.yml]: If v1.4 supports it, then yes. :)

**Comment by arkgil** [comment]
@josevalim I think it's ready. Now the builds are run for Elixir 1.4-1.7 on all supported OTP versions.

File: .travis.yml (hunk 2 above)
Comment (josevalim) [reply .travis.yml]: I don't think we need the `elixir` and `otp_release` sections now, do we?

Comment (arkgil) [reply .travis.yml]: We do. In `matrix` I have listed only versions which are not contained in the `elixir` and `otp_release`. To be honest, I thought that's what you mean when you said that you like versions to be included explicitly (previously some versions were being excluded). 

Just to make sure - you wanted all of the {OTP, Elixir} pairs to be included in the `matrix`, correct?

Comment (josevalim) [reply .travis.yml]: > Just to make sure - you wanted all of the {OTP, Elixir} pairs to be included in the matrix, correct?

Oh, I see. Yes, apologies, I think it is clearer if they are all listed explicitly.

**Comment by arkgil** [comment]
I think we can merge this one :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**